node/src/accountant/mod.rs

```diff
@@ -1120,29 +1135,22 @@ impl Accountant {
             );
 
         match result {
-            Ok(scan_message) => self
-                .retrieve_transactions_sub_opt
-                .as_ref()
-                .expect("BlockchainBridge is unbound")
-                .try_send(scan_message)
-                .expect("BlockchainBridge is dead"),
-            Err(e) => {
-                e.log_error(
-                    &self.logger,
-                    ScanType::Receivables,
-                    response_skeleton_opt.is_some(),
-                );
-
-                if let Some(skeleton) = response_skeleton_opt {
-                    self.ui_message_sub_opt
-                        .as_ref()
-                        .expect("UiGateway is unbound")
-                        .try_send(NodeToUiMessage {
-                            target: MessageTarget::ClientId(skeleton.client_id),
-                            body: UiScanResponse {}.tmb(skeleton.context_id),
-                        })
-                        .expect("UiGateway is dead");
-                };
+            Ok(scan_message) => {
+                self.retrieve_transactions_sub_opt
+                    .as_ref()
+                    .expect("BlockchainBridge is unbound")
+                    .try_send(scan_message)
+                    .expect("BlockchainBridge is dead");
+                ScanReschedulingAfterEarlyStop::DoNotSchedule
+            }
+            Err(e) =>
+            // Any error here panics by design, so the return value is unreachable/ignored.
+            {
+                self.handle_start_scan_error_and_prevent_scan_stall_point(
```

Comment:
This name's huge. Pick one: `resolve_scan_error` or `handle_scan_error_and_reschedule`.